data/modifications/pick_advance/dialog.lua

```diff
@@ -0,0 +1,168 @@
+-- << pickadvance_dialog
+
+pickadvance = {}
+local pickadvance = pickadvance
+local wesnoth = wesnoth
+local ipairs = ipairs
+local table = table
+local T = wesnoth.require("lua/helper.lua").set_wml_tag_metatable {}
+local translate = wesnoth.textdomain "wesnoth"
+
+local function filter_false(arr)
+	local result = {}
+	for _, v in ipairs(arr) do
+		if v ~= false then
+			result[#result + 1] = v
+		end
+	end
+	return result
+end
+
+
+function pickadvance.show_dialog_unsynchronized(advance_info, unit)
+-- dialog exit codes --
+  local help_code = -4
+  local reset_code = -3
+  local single_unit_code = -1
+  local all_units_code = 1
+--    
```

Comment:
Nice idea to document dialog exit codes on top!
On a practical side, should we use tabs/spaces uniformly? These 4 lines are written with two spaces, below everything is in tabs.

Reply:
Thanks, fixed - by default my text editor is set to use spaces instead.